src/main/java/com/google/cloud/spanner/pgadapter/statements/SimpleParser.java

```diff
@@ -836,6 +836,53 @@ boolean skipCommentsAndLiterals() {
     }
   }
 
+  Long readIntegerLiteral() {
+    skipWhitespaces();
+    int startPos = pos;
+    if (skipNumericLiteral()) {
+      try {
+        return Long.valueOf(sql.substring(startPos, pos));
+      } catch (NumberFormatException ignore) {
+        return null;
+      }
+    }
+    return null;
+  }
+
+  boolean peekNumericLiteral() {
+    int originalPos = pos;
+    if (skipNumericLiteral()) {
+      pos = originalPos;
+      return true;
+    }
+    return false;
+  }
+
+  boolean skipNumericLiteral() {
+    skipWhitespaces();
+    int startPos = pos;
+    if (eatToken("+") || eatToken("-")) {
```

Comment:
Is there a possibility of both being present? Or do we need checks for malformed query?

Reply:
Other methods that call this method need to check the result (it returns null if it was not a valid integer). The convention in this class is to return null in case the token was not consistent with the thing that the caller wanted to parse, as it makes it easier to implement optional steps, e.g. `fetch 1 from my_cursor` and `fetch from my_cursor` are equivalent.

`+-` is not allowed at the beginning, and this prevents that from being accepted, as this line will only read one of the tokens (either `+` or `-`). The rest of this method then expects the following token to be a digit (or `inf`).

I've added a couple of tests to verify.